src/middlewared/middlewared/plugins/apps/ix_apps/query.py

```diff
@@ -86,12 +86,19 @@ def list_apps(
         # When we stop docker service and start it again - the containers can be in exited
         # state which means we need to account for this.
         state = 'STOPPED'
+        exited_containers = 0
         for container in workloads['container_details']:
-            if container['state'] == 'starting':
-                state = 'DEPLOYING'
+            container_state = ContainerState(container['state'])
+            if container_state == ContainerState.STARTING:
```

Comment:
Each one of these if branches instantiates a new instance of the ContainerState class. You iterate the enum and see if the container_state matches the value